lag/lag.yaml

```diff
@@ -4,13 +4,25 @@ components:
       x-include: 
       - '../common/common.yaml#/components/schemas/Named.Object'
       description: >-
-        The container for multiple LAG ports
+        The container for LAG (ports group) - aggregation of multiple LAG members (ports)
       type: object
       properties:
         ports:
           type: array
           items:
             $ref: '#/components/schemas/Lag.Port'
+
+        protocol:
+          $ref: '#/components/schemas/Lag.Ports.Protocol'
+        min_links:
+          description: >-
+            Specifies the mininum number of member interfaces that must be active
+            for the aggregate interface to be available
+          type: integer
+          format: uint16
+          minimum: 0
+          maximum: 65535
```

Comment:
protocol spec aside, is this testable by us in real life?  what is a reasonable max?

Reply:
There will be validation in the implementation,  based on presently supported max. For example, Athena would validate for max value of 4 for now, which will increase to 8 or more gradually. In a different implementation (example in IxNetwork) - the limit can be different. So, should we change the max limit to model, or set it to max in the model and update the validation limit in the implementation, which can change from time to time.

Reply:
will there ever be an implementation that implements a LAG with 65535 members?  If not, what is a sensible default?  16, 32, 64, 128?  What is it?

Reply:
I agree with a small reasonable default - it can always be expanded in the future.

Reply:
I have reduced the max value to 32 for now. But this field value is not the actual limit of LAG members. It just says how many members to be 'UP' to consider LAG is also in 'UP' state. The actual limit of the lag member should be in "ports" property, which is of type 'array' of 'port' object. @ajbalogh , can we add a limit here?

Reply:
@PrasenjitAdhikary you can add maxItems and minItems to the ports array but an implementation MUST return a run time error if it is unable to meet them.  Is minItems=1 and maxItems=8 appropriate?

Reply:
Ports limit minItems=1 and maxItems=32 looks reasonable. At present protocol, engine supports 4 ports. So, implementation will through validation error if the ports count is more than 4. We will enhance the validation limit as and when support for more ports is added in Athena / Protocol engine.